src/Lean/Elab/Tactic/Simp.lean

```diff
@@ -204,7 +204,7 @@ def elabSimpArgs (stx : Syntax) (ctx : Simp.Context) (simprocs : Simp.SimprocsAr
                 arg[0][0].getKind == ``Parser.Tactic.simpPost
             let inv  := !arg[1].isNone
             let term := arg[2]
-            match (← resolveSimpIdTheorem? term) with
+            match (← withRef arg do resolveSimpIdTheorem? term) with
```

Comment:
```suggestion
            match (← withRef term do resolveSimpIdTheorem? term) with
```

It seems more reasonable to use the `term` itself rather than the whole `arg` (which  potentially includes the inversion arrow).

Reply:
Overlooked the second suggestion, but just addressed that now, see below.